### PR TITLE
Scale repair times with damage

### PR DESCRIPTION
**Pull request by mugling**

### CODE REVIEW

**Comment by DangerNoodle** [comment]
Will the end result of this be that repairing minor damage at least takes less time than repair under the current system?

File: src/veh_interact.cpp [hunk 1]
```diff
@@ -71,7 +71,11 @@ player_activity veh_interact::run( vehicle &veh, int x, int y )
             time = vp->install_time( g->u );
             break;
         case 'r':
-            time = pt->is_broken() ? vp->install_time( g->u ) : ( vp->repair_time( g->u ) * pt->base.damage() );
+            if( pt->is_broken() ) {
+                time = vp->install_time( g->u );
+            } else {
+                time = vp->repair_time( g->u ) * double( pt->base.damage() ) / pt->base.max_damage();
```
Comment (mugling) [inline src/veh_interact.cpp]: But if `max_damage() == 0` then `damage() == 0` and the repair code block is never called!

**Comment by mugling** [comment]
> Will the end result of this be that repairing minor damage at least takes less time than repair under the current system?

Yes, typical reduction will be by 1/4. For items not specifying an explicit repair time the default will be 1/8 of install time.

File: src/veh_interact.cpp (hunk 1 above)
Comment (codemime) [inline src/veh_interact.cpp]: > the repair code block is never called!

Yes, it's assumed, but isn't guaranteed as the code evolves. I suppose this assertion deserves either an `assert` / clause / comment here, in `veh_interact::run()`. BTW `int vehicle_part::hp()` (called by `is_broken()`) has the very same issue already.

**Comment by DangerNoodle** [comment]
Ah, sounds like a good idea overall then, barring any coding oddities that I would not be able to easily spot.

**Comment by mugling** [comment]
@Coolthulhu note recent push (1a109e1c). @codemime's suggestion to explicitly document preconditions is reasonable. We should be less hesitant about using `assert`.

* A `debugmsg` means shouldn't happen, but we will try to salvage the situation
* An `assert` should never happen and is used to convert a generic error (segfault, divide by zero) into a more descriptive one

I'd prefer to debug an assertion with a line number as opposed to a random crash

**Pull request merged**